rclcpp/test/rclcpp/test_parameter.cpp

```diff
@@ -44,6 +44,9 @@ TEST_F(TestParameter, not_set_variant) {
   rclcpp::Parameter not_set_variant;
   EXPECT_EQ(rclcpp::PARAMETER_NOT_SET, not_set_variant.get_type());
   EXPECT_EQ("not set", not_set_variant.get_type_name());
+  std::stringstream ss;
+  ss << not_set_variant;
+  EXPECT_EQ("{\"name\": \"\", \"type\": \"not set\", \"value\": \"not set\"}", ss.str());
```

Comment:
I know you didn't change this, but this is weird thing to be returning when trying to stream `PARAMETER_NOT_SET`.  I'd find it much more intuitive if the stream operator threw an exception.

If you agree, my suggestion is that we leave this test as-is (to test the current behavior), but open a follow-up issue to fix it later.

Reply:
I'm not aware of many uses of throwing exceptions in the `<<` operator, but I guess it depends on how much of an error it assumed to be when trying to stream an unset parameter.

Follow-up issue for comment created here. https://github.com/ros2/rclcpp/issues/1348